cpp/bench/neighbors/knn.cuh

```diff
@@ -179,9 +179,8 @@ struct ivf_pq_knn {
   {
     index_params.n_lists = 4096;
     index_params.metric  = raft::distance::DistanceType::L2Expanded;
-
-    auto data_view = raft::make_device_matrix_view<const ValT, IdxT>(data, ps.n_samples, ps.n_dims);
-    index.emplace(raft::neighbors::ivf_pq::build(handle, index_params, data_view));
+    index.emplace(raft::neighbors::ivf_pq::build(
```

Comment:
We don't want to revert this back to the pointer based APIs. We should always be preferring the mdspan APIs now (eventually pointer APIs will be removed)